domain/personallearningplan/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/personallearningplan/service/GoalService.kt

```diff
@@ -97,33 +97,33 @@ class GoalService(
     val existingGoal = getGoal(prisonNumber, updatedGoalDto.reference)
     return goalPersistenceAdapter.updateGoal(prisonNumber, updatedGoalDto)
       ?.also {
-        goalEventService.goalUpdated(prisonNumber = prisonNumber, previousGoal = existingGoal, updatedGoal = it)
         updateNote(prisonNumber, it, updatedGoalDto.notes)
+        goalEventService.goalUpdated(prisonNumber = prisonNumber, previousGoal = existingGoal, updatedGoal = it)
       }
       ?: throw GoalNotFoundException(prisonNumber, goalReference).also {
         log.info { "Goal with reference [$goalReference] for prisoner [$prisonNumber] not found" }
       }
   }
 
   // if the note is different to the note before then update/create the note
-  private fun updateNote(prisonNumber: String, goal: Goal, updatedNote: String?) {
-    val existingNote = goalNotesService.getNotes(goal.reference)
-
-    when {
-      existingNote == null && updatedNote != null -> {
```

Comment:
Essentially this was the bug - the existing note entity was null (ie. no previous note on the goal), but the request DTO had an empty string (not null) for the updated note content; which then threw this hibernate error:

![image](https://github.com/user-attachments/assets/af8a1a8a-64ee-4abc-8353-50ce7fa0a825)